### PR TITLE
Wye 9 create feadback

### PR DESCRIPTION
**Pull request by shankisg**
Disclaimer: After giving lot of thoughts, It have done changes in models related to workshop feedback. Pls review it carefully.  I have done the change to make insertion/ retrieval of  feedback easy . 

Contains first part implementation for #9 
 Feature that have been implemented
- Dynamic generation of feedback form depending on question in WorkshopRatingValues.
- Form will display all the questions from WorkshopRatingValues models plus a comment box.
- The workshop url will throw 404, unless workshop is completed.

Sample url: http://127.0.0.1:8000/workshop/1/feedback/

Once this is merge, I will add pull request for next part that will contain
- Restricting organisation to add workshop unless they complete feedback
- Restricting presenter to take up next workshop unless they complete feedback
- Display message to presenter/organiser to share feedback.

### CODE REVIEW

**Comment by vnbang2003** [comment]
@shankisg Correct me if my understand is right. For feedback models you have added new fields to capture Feedbacktype to easy the query to fetch it.

**Comment by shankisg** [comment]
yes, plus added foreign key in WorkshopVoting model refers to  WorkshopFeedback model. 

Pls review, WorkshopVoting and WorkshopFeedback models

File: wye/workshops/models.py [hunk 1]
```diff
@@ -121,44 +128,78 @@ class WorkshopRatingValues(TimeAuditModel):
...
+
+        workshop_feedback = cls.objects.create(
+            workshop=workshop,
+            comment=comment,
+            feedback_type=feedback_type
+        )
+        WorkshopVoting.save_rating(workshop_feedback, **kwargs)
+
+
+class WorkshopVoting(TimeAuditModel):
```
Comment (vnbang2003) [inline wye/workshops/models.py]: Is this models to capture the vote values?

**Comment by shankisg** [comment]
WorkshopVoting will store the response a for each question and WorkshopFeedBack will store the comments.

Attached is the model diagram to have clear picture of model relationships. Pls let me know if it works.

feedback

**Comment by shankisg** [comment]
@vnbang2003
I think, let's got ahead with this implementation for V0.1, later if needed we can change the implementation.

**Comment by vnbang2003** [comment]
@shankisg Final query does it take care of more than one requester/presenter vote?

**Comment by shankisg** [comment]
@vnbang2003  yes, it does take care of that.

**Comment by kracekumar** [comment]
@shankisg Have a look at [junction feedback](https://github.com/pythonindia/junction/blob/master/junction/feedback/models.py). The proposed one misses question title.

**Comment by shankisg** [comment]
@kracekumar what is use of title, if we have already have the question.

**Comment by kracekumar** [comment]
The feedback isn't configurable. How will new questions be added in future
?

**Comment by shankisg** [comment]
Yes, you can.  Just go ahead and add a question in WorkshopRatingValues model for name, this question will automatically appear.

The model **name** attribute is bit miss leading, may be we can change it to **question**.

File: wye/base/constants.py [hunk 2]
```diff
@@ -49,6 +49,21 @@ class OrganisationType:
...
 
 
+@choices
+class FeedbackType:
+    _PRESENTER = [1, "Presenter"]
+    _ORGANISATION = [2, "Organisation"]
+
+
+@choices
+class WorkshopRatings:
```
Comment (kracekumar) [inline wye/base/constants.py]: Too many options, I would suggest have only 3 options.

Comment (kracekumar) [inline wye/base/constants.py]: Ignore this since it is merged.

**Pull request merged**